lib/src/metta/runner/mod.rs

```diff
@@ -819,12 +825,15 @@ impl<'input> RunContext<'_, '_, 'input> {
 
         // Resolve the module name into a loader object using the resolution logic in the pkg_info
         #[cfg(feature = "pkg_mgmt")]
-        self.load_module_recursive(mod_name)
+        self.load_module_recursive(mod_name, true)
     }
 
     /// Internal function used for recursive loading of parent modules by [Self::load_module]
+    /// Calling with `load_target == false` will ensure a module's parents are loaded without
+    /// loading the module itself, and the module's parent's [ModId] will be returned.  Passing
+    /// `load_target == true` will load the module and return its [ModId]
     #[cfg(feature = "pkg_mgmt")]
-    fn load_module_recursive(&mut self, mod_name: &str) -> Result<ModId, String> {
+    fn load_module_recursive(&mut self, mod_name: &str, load_target: bool) -> Result<ModId, String> {
```

Comment:
Adding `load_target` flag is probably ok while method is private, but I would replace call with `false` flag by pair of calls: `load_module(get_parent_mod_name(mod_name))`. Thus the code clearly states that parent module is loaded and explanation of the `load_target` semantics is not needed.

Reply:
At least I would rename flat into `load_parent` as it is exactly its semantics as far as I understand.

Reply:
Done in 79a24c6